keras_contrib/applications/densenet.py

```diff
@@ -501,10 +527,24 @@ def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_bl
     x = BatchNormalization(axis=concat_axis, gamma_regularizer=l2(weight_decay),
                            beta_regularizer=l2(weight_decay))(x)
     x = Activation('relu')(x)
-    x = GlobalAveragePooling2D()(x)
+    if pooling is not None:
+        x = GlobalAveragePooling2D()(x)
 
-    if include_top:
-        x = Dense(nb_classes, activation='softmax', kernel_regularizer=l2(weight_decay), bias_regularizer=l2(weight_decay))(x)
+    if include_top and pooling is not None:
```

Comment:
The change you suggest makes sense. I would actually argue that a proper description of all 4 cases be included in the documentation of DenseNetFCN as examples, because I feel users will get confused between the interaction of pooling and include top. 

In the final places, pooling kind of refers to GlobalAveragePooling, whereas in earlier stages it refers to intermediate pooling steps (MaxPooling or AveragePooling). A good example documentation will go a long way in removing the confusion.